application/comit_node/src/http_api/rfc003/with_swap_types.rs

```diff
@@ -65,3 +65,67 @@ macro_rules! with_swap_types {
         }
     }};
 }
+
+macro_rules! _match_role_bob {
+    ($role:ident, $fn:tt) => {
+        #[allow(clippy::redundant_closure_call)]
+        match $role {
+            RoleKind::Bob => {
+                #[allow(dead_code)]
+                type Role = Bob<AL, BL, AA, BA>;
+                $fn()
+            }
+            _ => unimplemented!(),
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! with_swap_types_bob {
```

Comment:
is it a one-time use macro? is there much point?

Reply:
It saves us about 60 lines of duplicated code for every `(alpha_ledger, beta_ledger, alpha_asset, beta_asset)` tuple.

Reply:
Ok I understand know, should have said that the macro repeats the passed arguments... 😄